AppointmentDetailsExtractor.cs

```diff
@@ -0,0 +1,69 @@
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace TwilioSmsScheduler;
+
+public static class AppointmentDetailsExtractor
+{
+    public static AppointmentDetails Extract(string smsBody, string timeZone, string senderPhoneNo)
```

Comment:
This logic is quite brittle, so for the sake of the demo it is fine, but make sure to note to the user they should be using AI models to extract this information.